src/python/tests/test_maf_to_fasta.py

```diff
@@ -0,0 +1,133 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `maf_to_fasta.py` script.
+
+Typical usage example::
+
+    $ pytest test_maf_to_fasta.py
+
+"""
+
+from contextlib import nullcontext as does_not_raise
+import filecmp
+from pathlib import Path
+import re
+from typing import ContextManager, Iterable, List, Optional, Pattern
+
+import pytest
+from pytest import raises
+
+from ensembl.compara.filesys.dircmp import PathLike
+from ensembl.compara.hal import maf_to_fasta
+
+
+class TestMafToFasta:
+    """Tests script ``maf_to_fasta.py``"""
+
+    ref_file_dir = None  # type: Path
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self) -> None:
+        """Loads necessary fixtures and values as class attributes."""
+        type(self).ref_file_dir = pytest.files_dir / 'hal_alignment'
+
+    @pytest.mark.parametrize(
+        "genome_names, exp_output, expectation",
+        [
+            (['genomeA.1', 'genomeB.1'], re.compile('^(?P<genome>genomeA\\.1|genomeB\\.1)[.](?P<seqid>.+)$'),
+                does_not_raise()),
+            (iter(['genomeA.1', 'genomeB.1']), None, raises(ValueError)),
+            (['genomeA', 'genomeA.1'], None, raises(ValueError)),
+            ([], None, raises(ValueError))
+        ]
+    )
+    def test_compile_maf_src_regex(self, genome_names: Iterable[str], exp_output: Pattern[str],
+                                   expectation: ContextManager) -> None:
+        """Tests :func:`maf_to_fasta.compile_maf_src_regex()` function.
+
+        Args:
+            genome_names: The genome names expected to be in the input MAF file.
+            exp_output: Expected return value of the function.
+            expectation: Context manager for the expected exception. The test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            # pylint: disable-next=no-member
+            obs_output = maf_to_fasta.compile_maf_src_regex(genome_names)
```

Comment:
I'm wondering how much "local" the other functions in this module are, and if it would be worth adding them in the `__all__` clause (it would, among other things, get rid of the `pylint` comment). This one I'm more conflicted, but not so much about the next one (`map_uint_to_path`). What do you think?

Reply:
My thinking was mainly that it's easier to add a function to the `__all__` clause than to remove it. Also, I was thinking that if `map_uint_to_path` turns out to be useful outside of this module, it would probably belong somewhere else anyway (e.g. `utils` module).